gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/BundleMapperService.java

```diff
@@ -243,12 +245,30 @@ private Organization getPatientOrganization(List<? extends DomainResource> agent
             .orElse(null);
     }
 
-    private RCMRMT030101UK04EhrFolder getEhrFolder(RCMRIN030000UK06Message xmlMessage) {
-        return xmlMessage.getControlActEvent().getSubject().getEhrExtract().getComponent().get(0).getEhrFolder();
+    private RCMRMT030101UK04EhrFolder getEhrFolder(RCMRIN030000UKMessage xmlMessage) {
+        if (xmlMessage instanceof RCMRIN030000UK07Message) {
```

Comment:
What's the plan for this `if` statements (and others), they look a bit grim are them temprary?